internal/controllers/common/resource_definitions/resource_definitions.go

```diff
@@ -303,6 +353,70 @@ func NewPodForCR(cr *operatorv1beta1.Cryostat, specs *ServiceSpecs, imageTags *I
 	}
 }
 
+const reportsPort = 10000
+
+func NewPodForReports(cr *operatorv1beta1.Cryostat, imageTags *ImageTags) *corev1.PodSpec {
+	resources := corev1.ResourceRequirements{}
+	if cr.Spec.ReportOptions != nil {
+		resources = cr.Spec.ReportOptions.ResourceRequirements
+	}
+	cpus := int64(1)
+	if requests := resources.Requests; requests != nil {
+		if cpu := requests.Cpu(); cpu != nil {
+			cpus = cpu.Value()
+		}
+	}
+	if limits := resources.Limits; limits != nil {
+		if cpu := limits.Cpu(); cpu != nil {
+			cpus = cpu.Value()
+		}
+	}
```

Comment:
For this case yes. If the limits are present they should be greater than or equal to the requests, so I think this is effectively equivalent to taking the maximum of the two values. Whichever is the maximum is what we tell the JVM is how much CPU it can expect to be able to use. Maybe that should ignore the limits and only use the requests?